controllers/perconaservermysql_controller.go

```diff
@@ -297,21 +300,43 @@ func (r *PerconaServerMySQLReconciler) doReconcile(
 	return nil
 }
 
+func telemetryEnabled() bool {
+	value, ok := os.LookupEnv("DISABLE_TELEMETRY")
+	if ok {
+		return value != "true"
+	}
+	return true
+}
+
+func versionUpgradeEnabled(cr *apiv1alpha1.PerconaServerMySQL) bool {
+	return cr.Spec.UpgradeOptions.Apply != "" &&
+		cr.Spec.UpgradeOptions.Apply != apiv1alpha1.UpgradeStrategyDisabled &&
+		cr.Spec.UpgradeOptions.Apply != apiv1alpha1.UpgradeStrategyNever
+}
+
 func (r *PerconaServerMySQLReconciler) reconcileVersions(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
-	if cr.Spec.UpgradeOptions.Apply == "" ||
-		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyDisabled ||
-		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyNever {
+	if !(versionUpgradeEnabled(cr) || telemetryEnabled()) {
 		return nil
 	}
 
 	log := logf.FromContext(ctx).WithName("reconcileVersions")
 
-	version, err := vs.GetVersion(ctx, cr, r.ServerVersion)
-	if err != nil {
-		log.Info("Failed to get versions, using the default ones", "error", err)
+	if telemetryEnabled() && (!versionUpgradeEnabled(cr) || cr.Spec.UpgradeOptions.VersionServiceEndpoint != vs.GetDefaultVersionServiceEndpoint()) {
+		_, err := vs.GetVersion(ctx, cr, vs.GetDefaultVersionServiceEndpoint(), r.ServerVersion)
+		if err != nil {
+			log.Error(err, "failed to send telemetry to "+vs.GetDefaultVersionServiceEndpoint())
+		}
+	}
+
+	if !versionUpgradeEnabled(cr) {
 		return nil
 	}
 
+	version, err := vs.GetVersion(ctx, cr, cr.Spec.UpgradeOptions.VersionServiceEndpoint, r.ServerVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to get versions, using the default ones")
```

Comment:
In fact `"failed to get versions, using the default ones"` should be a log, and for the error just what failed. Something like this:
```
log.Info("failed to get versions, using the default ones")
return errors.Wrap(err, "failed to get versions")
```

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/291/commits/cf68585fb494946f31a373103334d5c0c0a1a51f